### DISCUSSION TITLE
How to improve actions?

### DISCUSSION THREAD

**Discussion by sagostine** [UNC-Project-Open-AAC/OS-DPI#208] [Help]
Hey y'all! I have a question around actions....I figured it would be easiest to just show you vs. trying to type it out so see the video I have attached.

[Actions question.webm](https://github.com/UNC-Project-Open-AAC/OS-DPI/assets/123391336/3584e142-021c-429e-8bca-68b533f78a21)

I have also attached the osdpi file if you want to mess around with my excel sheet!
[Introception_Grid_6_30_23.zip](https://github.com/UNC-Project-Open-AAC/OS-DPI/files/11923021/Introception_Grid_6_30_23.zip)

Susie

**Comment by gbishop** [comment]
It feels like a bug in OS-DPI to me. I'm watching the monitor and it shows
that #display has the value itchy on the Home button. But that is not in
the content. I'll have to track it down.

**Comment by sofia-bg** [comment]
I don't think it is a bug. 

From what I see you want to combine those actions, because you want each of those buttons to do both actions at the same time (fill the display AND change the page). At least when I watch it looks like nearly all of them result in a change page. So by having them separate the problem is that the system only does the one that comes first. 

It might be true that that top level you don't want the label to fill, so that might be a second action based on some attribute of them that tells the system only to have those change the page and not fill the display. 

Does that make sense at all?

**Reply by sofia-bg** [reply]
or maybe it was a bug? because based on your spread sheet I see I am wrong and that things that have a display only have a display and things that change page do not have a display. So ignore me :)

**Comment by gbishop** [comment]
Susie turned up a couple of bugs with her design. First we didn't support JFIF image format (which I didn't know existed). Second, we had a very subtle bug that caused fields to persist on buttons when they shouldn't. I think I have fixed them both.
